src/rdkafka_partition.c

```diff
@@ -3255,6 +3276,21 @@ rd_kafka_topic_partition_t *rd_kafka_topic_partition_list_find_topic(
                 return &rktparlist->elems[i];
 }
 
+/**
+ * @returns the first element that matches \p topic_id, regardless of partition.
+ */
+rd_kafka_topic_partition_t *rd_kafka_topic_partition_list_find_topic_by_id(
```

Comment:
Better to name this `rd_kafka_topic_partition_list_get_topic_by_topic_id`

Reply:
Used the same naming of the existing
`rd_kafka_topic_partition_list_find_topic`
maybe I can rename that to `rd_kafka_topic_partition_list_find_topic_by_name`